python/astro_metadata_translator/tests.py

```diff
@@ -149,8 +150,26 @@ def assertObservationInfoFromYaml(self, file, dir=None, check_wcs=True,  # noqa:
             inconsistent.
         """
         header = read_test_file(file, dir=dir)
-        self.assertObservationInfo(header, filename=file, check_wcs=check_wcs,
-                                   wcs_params=wcs_params, **kwargs)
+
+        try:
```

Comment:
Looking at this again (thanks for simplifying it) how do you feel about reducing the code duplication? Something like:

```python
astropy_header = Header()
for key, val in header.items():
    ...

for hdr, msg in ((header, "dict-like"), (astropy_header, "Astropy header")):
    try:
        self.assertObservationInfo(hdr, ...)
    except AssertionError as e:
        raise(f"ObservarionInfo derived from {msg} type is inconsistent.") from e
```
?

Reply:
You could even give up on the `msg` variable and just put in `type(hdr)` instead...